main/src/ca/uwaterloo/flix/language/CompilationMessageKind.scala

```diff
@@ -36,6 +35,28 @@ sealed trait CompilationMessageKind {
     case TypeError => "Type Error"
     case WeederError => "Syntax Error"
   }
+
+  /**
+    * Returns the next compilation message kind in the phase order.
+    *
+    * Returns `None` if this is the last phase.
+    */
+  def next: Option[CompilationMessageKind] = this match {
```

Comment:
This seems overly repetitive. Why not just make a list, do sliding window, and get a map out of it?

Reply:
I also thought about that. Really its a total order-- but I need the ability to get the next element. So its like the Discrete type class (or the PartialPrev, PartialNext)-- which we lack. So ultimately I don't know. I think if I am smart I make it into a total pattern match without wildcard case-- that way its better than a list since it breaks if a new phase is added.